rfcs/2024-08-21-motion-presets.md

```diff
@@ -0,0 +1,555 @@
+---
+Start Date: 21-08-2024
+RFC PR: (leave this empty if no PR yet)
+Blade Issue: (leave this empty if no issue yet)
+---
+
+# Motion Presets RFC <!-- omit in toc -->
+
+### Table Of Contents <!-- omit in toc -->
+
+- [Summary](#summary)
+- [Basic Example](#basic-example)
+- [Motivation](#motivation)
+- [Detailed Design](#detailed-design)
+- [Drawbacks/Constraints](#drawbacksconstraints)
+- [Alternatives](#alternatives)
+- [Adoption strategy](#adoption-strategy)
+- [How do we educate people?](#how-do-we-educate-people)
+- [Open Questions](#open-questions)
+- [References](#references)
+
+# Summary
+
+# Basic Example
+
+```jsx
+import { Fade, Card, CardBody } from '@razorpay/components/blade';
+
+<Fade isVisible={showCard}>
+  <Card>
+    <CardBody>{/* Blade Card */}</CardBody>
+  </Card>
+</Fade>;
+```
+
+Checkout full [API Decisions for Motion Presets](#api-decisions)
+
+# Motivation
+
+You might have seen our previous [RFC on Motion Foundations](./2022-03-22-motion-rfc.md) we wrote in 2022. In that RFC we defined token structure and foundational principles for motion.
+
+As our org grows, we believe our products need to go a bit beyond the functionality. How do we make sure that our consumers really love our products? How do we add that cherry on top, those sprinkles, that pineapple on pizza?
+
+<img src="./images/motion-presets/motion-presets-meme.png" height="200px" /> <img src="./images/motion-presets/motion-presets-consumers-love.gif" height="200px" />
+
+To solve for consumer delight, the blade team is working on simplifying adding motion in our consumer applications and introduce consistent animations to our product.
+
+You can check our detailed design proposal on [Motion Refresh](https://docs.google.com/document/d/1Vj2E0v2IdLSHuJNRrmX7X-BFSqGpXMFU0jj7LLfoZGk/edit?usp=sharing) (Only accessible to Razorpay Employees)
+
+# Detailed Design
+
+For building presets, we have to figure out few things like
+
+- [API Decision for Motion Presets](#api-decisions)
+- [Libarary to use for Animations](#library-comparison)
+
+## API Decisions
+
+> [!NOTE]
+>
+> The API decisions here are only there to give some basic idea on the structure and usage. More accurate props will be updated here later once they are finalised in design.
+
+### Entry / Exit Animation Presets
+
+#### Fade
+
+<table>
+<tr>
+  <td>
+
+```jsx
+import { Fade } from '@razorpay/blade/components';
+
+<Fade isVisible={showCard}>
+  <Card>
+    <CardBody>
+      <Text>Fade In/Out Card</Text>
+    </CardBody>
+  </Card>
+</Fade>;
+```
+
+```ts
+type FadeProps = {
```

Comment:
Should we provide atleast the `delay` prop to every primitive? 

So that consumers can choose to delay the animation run by x duration?

Reply:
These props are not finalized. Only structure is. Will add props depending on discussion with design

Reply:
On the same note, should we have tokens for `duration` if we don't allow explicit values? Certain use cases might need animations to run for different times. I understand it won't be an issue for looped ones, but for non-looped ones, this will help

Reply:
Atleast from design-side we're saying that the enter and exit durations should have same duration always. Do you know any usecase where a single duration won't work? we can also add this prop in future if usecase comes since this anyways won't be a breaking change then.